sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -815,13 +816,169 @@ public void validate(PipelineOptions options) {
 
       Pipeline p = input.getPipeline();
       final Coder<T> coder = inferCoder(p.getCoderRegistry());
+
       if (getMethod() == Method.BQ_PARALLEL_READ) {
-        return p.apply(org.apache.beam.sdk.io.Read.from(BigQueryParallelReadTableSource.create(
-            getTableProvider(),
-            getRowProtoParseFn(),
-            coder,
-            getBigQueryServices(),
-            getReadSessionOptions())));
+        // When using Method.BQ_PARALLEL_READ to read directly from a table, there are no temporary
```

Comment:
The current function is getting to be way too large, probably time to split it up into smaller functions, or factor out sub-transforms, or make some anonymous classes named.

Reply:
Done.